### PR TITLE
SOLID Principles refactoring👨‍💻 

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR we present SOLID Principles refactoring🚀 :
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Requirements
- Setup ruby workflow and linters 🔍.
- use GitFlow 🌊.
- Make sure `YAGNI`, `KISS`, and `DRY` rules and Ruby style guide are respected. 🔯 
- Follow `SOLID` principles. 🧱 
- Pay special attention to the `Single-responsibility` Principle.🥼 
         - Make sure that the role of `main.rb` is only to start your code and ask the user for the initial choice of options (or to quit).
         - Extract some pieces of logic into separate class(es).

## Achievements
- Setup ruby workflow and linters 🔍.✅
- use GitFlow 🌊.✅
- Make sure `YAGNI`, `KISS`, and `DRY` rules and Ruby style guide are respected. 🔯 ✅
- Follow `SOLID` principles. 🧱 ✅
- Pay special attention to the `Single-responsibility` Principle.🥼 ✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: assignment.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module Assignment
+  #   attr_accessor :classroom_assignment, :student_assignment
```
Comment (OyePriscilla) [inline assignment.rb]: - [ ] Please kindly remove this commented code to make your codebase clean. 👍

File: classroom.rb [hunk 2]
```diff
@@ -6,8 +8,9 @@ def initialize(label)
...
 
-  def add_student(student)
-    students << student
-    student.classroom = self
-  end
+  include Assignment
+  # def add_student(student)
+  #   students << student
+  #   student.classroom = self
+  # end
```
Comment (OyePriscilla) [inline classroom.rb]: - [ ] Please kindly remove this commented code to make your codebase clean. 👍

File: student.rb [hunk 3]
```diff
@@ -18,8 +21,9 @@ def play_hooky
...
 
-  def classroom=(classroom)
-    @classroom = classroom
-    classroom.students.push(self) unless classroom.students.include?(self)
-  end
+  include Assignment
+  # def classroom=(classroom)
+  #   @classroom = classroom
+  #   classroom.students.push(self) unless classroom.students.include?(self)
+  # end
```
Comment (OyePriscilla) [inline student.rb]: - [ ] Please kindly remove this commented code to make your codebase clean. 👍

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @gisachris and @kessie2862 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Refactored code snippet ✔️
`main.rb` was used to start app ✔️

## Required Changes ♻️

- [ ] In respect to `Single Responsibility Principle`, a class should have a single purpose and should be focused on doing that one thing well. Kindly take note the this methods do not follow `SRP `because, `all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.`

Untitled

## **Kindly implement this for **create_student, create_teacher, create_book and create_rentals methods**

Kindly reference [this link as a refresher](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 3 above)
Comment (gisachris) [reply student.rb]: Thankyou very much 👍 @OyePriscilla. We will implement the required changes and request for another review.

**Comment by gisachris** [comment]
## IN This PR We present Changes♻️ to SOLID Principles refactoring👨‍💻:
 
Below are some of my implementations as well as going over the Required Changes 🤝 .
## Changes Required ♻️ 
- Bring all Input & input Validation techniques to the `Main.rb` file as requested [here](https://github.com/gisachris/school-library-app/pull/5#pullrequestreview-1566840612)
- Remove clustering commented out `#` code snippets from the codeBase⛓️ 

## Achievements
-  Bring all Input & input Validation techniques to the `Main.rb` ✅ 
- Remove clustering commented out `#` code snippets from the codeBase⛓️  ✅ 

## Conclusion
**We are looking forward to hearing 👂  from you The CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

**Review by DammyShittu — APPROVED** [review]
Hi @gisachris and @kessie2862,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**